lib/semantics/resolve-names.cc

```diff
@@ -332,13 +329,14 @@ class ImplicitRulesVisitor : public DeclTypeSpecVisitor {
   }
 
 protected:
-  void PushScope();
-  void PopScope();
-  void ClearScopes() { implicitRules_.reset(); }
+  void BeginScope(const Scope &);
+  void SetScope(const Scope &);
 
 private:
+  // scope -> implicit rules for that scope
```

Comment:
Why not use a data member in `Scope`?  Is it because this mapping can change, or is emphemeral?

Reply:
The latter. I want to avoid putting things in Scope that aren't needed after semantic analysis.